File: Programmers/LEVEL_2/c++/쇠막대기.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(string arrangement) {
    int answer = 0;
    stack<int> s;

    for (int i = 0; i < arrangement.size(); i++)
    {
        char ch = arrangement[i];
        char ch2 = arrangement[i - 1];
        if (ch == '(')
        {
            s.push(ch);
        }
        else if (ch == ')')
        {
            s.pop();
            if (ch2 == '(')
            {
                answer += s.size();
            }
            else if (ch2 == ')')
            {
                answer += 1;
            }
        }
    }

    return answer;
}

File: Programmers/LEVEL_2/쇠막대기.js
function solution(arrangement) {
  var answer = 0;

  let stack = [];

  for (let i in arrangement) {
    if (arrangement[i] == "(") {
      stack.push(i);
    } else {
      if (arrangement[i - 1] == "(") {
        stack.pop();
        answer += stack.length;
      } else if (arrangement[i - 1] == ")") {
        stack.pop();
        answer += 1;
      }
    }
  }

  return answer;
}
